core/drivers/crypto/caam/hal/common/hal_sm.c

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2019, 2023 NXP
+ */
+#include <caam_sm.h>
+#include <caam_common.h>
+#include <caam_hal_sm.h>
+#include <caam_io.h>
+#include <caam_status.h>
+#include <kernel/boot.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+#include <registers/sm_regs.h>
+#include <registers/version_regs.h>
+#include <stdint.h>
+#include <tee_api_defines.h>
+#include <tee_api_types.h>
+
+/*
+ * Issue a Secure Memory Command to the @page and @partition.
+ * Returns the command status when completed
+ *
+ * @jr_base    JR virtual base address
+ * @page       Memory Page
+ * @partition  Partition
+ * @cmd        Command to sent
+ */
+static uint32_t issue_cmd(vaddr_t jr_base, unsigned int page,
+			  unsigned int partition, uint8_t cmd)
+{
+	uint32_t status = 0;
+	uint64_t timeout_ref = timeout_init_us(10000);
+
+	assert(jr_base);
+
+	/* Send cmd */
+	io_caam_write32(jr_base + SM_SMCR, SM_SMCR_PAGE(page) |
+						   SM_SMCR_PRTN(partition) |
```

Comment:
Strange indentation.

Reply:
This is formatted by clang-format

Reply:
I guess this is one of the times when it's impossible to tune `clang-format` to follow the coding guidelines.
Please fix the indentation.

Reply:
I don't really see the point of merging a `.clang-format` if we need to go over all the code format again. I understand maintainers want to keep the code format coherent, but tolerating one or two rules in favor of `clang-format` could greatly ease the review process.
At NXP, we've added clang-format to our CI, and we no longer care about the code format. Of course, there are a few quirks sometimes that require manual intervention, but the amount of time spent on fixing it is insignificant compared to the time we spent reviewing the code format.
Anyway, I will fix the formatting as you ask.